lambdas/service/app.py

```diff
@@ -655,204 +655,256 @@ def repository_search(entity_type: str, item_id: str):
         raise NotFoundError(msg=e)
 
 
-@app.route('/repository/files', methods=['HEAD', 'GET'], cors=True)
-@app.route('/repository/files/{file_id}', methods=['GET'], cors=True)
+generic_object_spec = schema.object(additional_properties=True)
+array_of_object_spec = schema.array(generic_object_spec)
+hit_spec = schema.object(
+    additional_properties=True,
+    protocols=array_of_object_spec,
+    entryId=str,
+    samples=array_of_object_spec,
+    specimens=array_of_object_spec,
+    cellLines=array_of_object_spec,
+    donorOrganisms=array_of_object_spec,
+    organoids=schema.array(str),
+    cellSuspensions=array_of_object_spec,
+)
+
+page_spec = schema.object(
+    hits=schema.array(hit_spec),
+    pagination=generic_object_spec,
+    termFacets=generic_object_spec
+)
+
+
+def filters_param_spec(facets):
+    array_pair_spec = schema.array({}, minItems=2, maxItems=2)
+    return params.query(
+        'filters',
+        schema.optional(
+            application_json(
+                schema.object(
+                    # A more correct schema would be:
+                    # schema.object(**{
+                    #   facet: schema.optional({
+                    #       'oneOf': [
+                    #           schema.object(**{rel: schema.array({})})
+                    #           for rel in ['is', 'contains', 'within', 'intersects']
+                    #       ]
+                    #   })
+                    #   for facet in facets
+                    # })
+                    # But Swagger does not informatively render the `oneOf`
```

Comment:
But the Swagger doc explicitly mentions that you can use these qualifiers: https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/ 

Is it possible that when you say "render", you're referring to the generated example instead of the how the schema is rendered? The unmerged PR I was referring to is about the generated examples: https://github.com/swagger-api/swagger-ui/pull/5530#issuecomment-531365696

If so, we should definitely use them. And I think you may have missed

https://github.com/DataBiosphere/azul/pull/1647#discussion_r406567769

addressing which potentially side-steps all of this by providing a hand-made example.

Reply:
I've implemented the manual examples. Oddly, Apidev reports structural errors, claiming that the manual example feature is incompatible with the `content` parameter. The rendered UI seems to work fine though.